### ISSUE TITLE
Code review for assignment 1

### ISSUE THREAD

**Issue by ES-UoL** [LMHedges/CMP1903M-A01-2223#2]
Documentation:

+ Good use of comments throughout which makes it easy to follow the code. 
+ The code looks well written and is fully functional to the requirements that were outlined in the assignment brief.
+ Good use of switches to determine the card identity.
+ The inclusion of the testing menu is great as it helps test specific functions of the code without having to run through everything which improves readability and testing efficiency.

- There were a few spelling mistakes for example "Dimonds" should be "Diamonds". It is important to make sure that words in the code are spelt correctly to maintain professional standards to the client's requirements as well as to ensure proper integration with the client's code as mentioned in the brief scenario.

Exception Handling:

+ Good use of exception handling to ensure that the test menu does not crash upon entering a number out of the range of the specified options.
+ Exception handling is also present to deal with a situation in which a letter is entered instead of a number in the testing menu. This also prevents a crash from occurring.
    
- No information is provided when the exceptions are caught to explain the problem with the input on the main testing menu. This lack of feedback may cause confusion for the user. However there is feedback with invalid input in the shuffle and deal submenus which is great but could be more specific to the cause of the error rather than simply saying "Not valid input, please try again".
- No validation is present when the Card class properties are set which may cause problems with future development since new cards could be created with values beyond what is expected.
   
Conclusion:

In conclusion, the improvements that I would suggest based on my review would be to correct any spelling mistakes, and ensure that everything is validated with sufficient feedback to the user. Overall, great work! LGTM.

**Comment by LMHedges** [comment]
Thank you for the code review, I will look into implementing the changes that you have suggested.
